package Exception_Programs;
import java.util.Scanner;

class nonintresultexception  extends ArithmeticException{ // Since this is Superclass for 
	public nonintresultexception(String i) {
		super(i);
	}
	
}

public class ExceptionAssignmentQues1 {
	
	
	public static void main(String args[]) {
			 int q = 0;
			 Scanner myobj = new Scanner(System.in);
			 System.out.println("Enter an interger: ");
			 int i = myobj.nextInt();
			 System.out.println("Enter another Interger: ");
			 int j = myobj.nextInt();			 
			 
			 try {
				
			 q = i/j;
//			 quotient = a + " ";
//			 quotient = Integer.parseInt(quotient);
//			 System.out.println(quotient);
			 
			 }
			 
			 catch (nonintresultexception ae) {
				 String str = null;
				 str = String.format("Integer %2d divided by %2d does not give an integer", i, j);	
				 System.out.println(str);
			 }
			 finally {  
			System.out.println(q);}
			
			
			
	}
	
	
}
